### PR TITLE
rewrite finetuning data preprocessing with static shape for Gaudi2.

### PR DESCRIPTION
**Pull request by lkk12014402**
## Type of Change

feature/bug enhancement that supports static shape input for gaudi2

## Description

rewrite finetuning data preprocessing with static shape for Gaudi2

fix issue: [#1201 ](https://github.com/intel/intel-extension-for-transformers/issues/1201)

### CODE REVIEW

**Comment by lkk12014402** [comment]
for llava finetuning stage, 4-cards gaudi2
lora zero2
image

lora zero3


full parameters zero2

full parameters zero3

**note**: 
- there are still unresolved dynamic shapes in the input
- when running on Gaudi2, should set `--pad_max True` in your script

File: intel_extension_for_transformers/neural_chat/examples/finetuning/multi_modal/scripts/finetune.sh [hunk 1]
```diff
@@ -15,7 +15,8 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-deepspeed train.py \
+PT_HPU_MAX_COMPOUND_OP_SIZE=10 DEEPSPEED_HPU_ZERO3_SYNC_MARK_STEP_REQUIRED=1 \
+python3 ./gaudi_spawn.py --use_deepspeed --world_size 4 \
```
[the remark this answers is not in the document]
Comment (lkk12014402) [reply intel_extension_for_transformers/neural_chat/examples/finetuning/multi_modal/scripts/finetune.sh:19]: Thanks! fixed~

**Review by VincyZhang — APPROVED** [review]

File: intel_extension_for_transformers/neural_chat/examples/finetuning/multi_modal/train.py [hunk 2]
```diff
@@ -108,6 +113,25 @@ def train():
     parser = transformers.HfArgumentParser(
         (ModelArguments, DataArguments, TrainingArguments))
     model_args, data_args, training_args = parser.parse_args_into_dataclasses()
+
+    # Setup logging
+    logging.basicConfig(
+        format="%(asctime)s - %(levelname)s - %(name)s - %(message)s",
+        datefmt="%m/%d/%Y %H:%M:%S",
+        handlers=[logging.StreamHandler(sys.stdout)],
```
Comment (dillonalaird) [inline intel_extension_for_transformers/neural_chat/examples/finetuning/multi_modal/train.py:121]: Missing `import sys`

**Review by dillonalaird — COMMENTED** [review]
Ran the script on my end and everything worked minus that sys import, thanks again for the PR!

File: intel_extension_for_transformers/neural_chat/examples/finetuning/multi_modal/train.py (hunk 2 above)
Comment (lkk12014402) [reply intel_extension_for_transformers/neural_chat/examples/finetuning/multi_modal/train.py:122]: Thanks for the comment, fixed!

**Review by hshen14 — APPROVED** [review]

**Pull request merged**